module-system/sov-modules-stf-template/src/app_template.rs

```diff
@@ -163,107 +165,130 @@ where
                 }
             };
 
-            match RT::decode_call(tx.runtime_msg()) {
-                Ok(msg) => {
-                    let ctx = C::new(sender_address.clone());
-                    let tx_result = self.runtime.dispatch_call(msg, &mut batch_workspace, &ctx);
-
-                    self.runtime
-                        .post_dispatch_tx_hook(&tx, &mut batch_workspace)
-                        .expect("Impossible happened: error in post_dispatch_tx_hook");
-
-                    let tx_effect = match tx_result {
-                        Ok(_) => TxEffect::Successful,
-                        Err(e) => {
-                            debug!(
-                                "Tx 0x{} was reverted error: {}",
-                                hex::encode(raw_tx_hash),
-                                e
-                            );
-
-                            // The transaction causing invalid state transition is reverted but we don't slash and we continue
-                            // processing remaining transactions.
-                            batch_workspace = batch_workspace.revert().to_revertable();
-                            TxEffect::Reverted
-                        }
-                    };
+            let ctx = C::new(sender_address.clone());
+            let tx_result = self.runtime.dispatch_call(msg, &mut batch_workspace, &ctx);
 
-                    let receipt = TransactionReceipt {
-                        tx_hash: raw_tx_hash,
-                        body_to_save: None,
-                        events: batch_workspace.take_events(),
-                        receipt: tx_effect,
-                    };
+            // TODO: Do we want to call it before revert in case of error?
+            // TODO: Replace expect with proper handling
```

Comment:
Should I address both of these in this PR or offload to separate issue?

Reply:
Yes I think we should revert.

Reply:
Yes, this will be discussed as part of https://github.com/Sovereign-Labs/sovereign-sdk/issues/421